sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/V1Table.scala

```diff
@@ -58,7 +59,7 @@ private[sql] case class V1Table(v1Table: CatalogTable) extends Table {
 
   override lazy val properties: util.Map[String, String] = addV2TableProperties(v1Table).asJava
 
-  override lazy val schema: StructType = v1Table.schema
+  override lazy val schema: StructType = CharVarcharUtils.getRawSchema(v1Table.schema, SQLConf.get)
```

Comment:
why is this change?

Reply:
when v2 command work with v1 table. when we resolve the table. we load the table by V2SessionCatalog, then create `V1Table(sessionCatalog.getTableMetadata)`, here when `getTableMetadata`, we will replace char/varchar with string in schema. So `V1Table(t)` t is alway with string type although type is char/varchar. So re-construct the schema of V1Table.

Reply:
maybe we can remove [#L496](https://github.com/apache/spark/blob/ef837ca71020950b841f9891c70dc4b29d968bf1/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala#L496)

Reply:
I think we should restore the raw table schema in SHOW CREATE TABLE, not here.

Reply:
other v2 command works v1 table. maybe encounter this problem. but I will update as you said. thank you very much.

Reply:
done.